### DISCUSSION TITLE
2.0.2 is out!

### DISCUSSION THREAD

**Discussion by Gissio** [Gissio/radpro#120] [Announcements]
Rad Pro 2.0.2 is out, bringing several minor enhancements.

The instantaneous rate averaging algorithm has been refined, eliminating statistical quirks that could lead to imprecise readings. Also, the averaging response curves have been improved, delivering enhanced precision with J613/J614 tubes.

To minimize false alarms, the instantaneous rate alarm now triggers only when the confidence interval falls below 75%.

Devices with a color screen now feature a subtle shadow between the title bar and the main screen, enhancing readability.

Numerous minor issues with the documentation have also been addressed.

Enjoy!

**Comment by dc1rdb** [comment]
Thanks for the new release!
Fnirsi GC-01 with Geehy processor here.

Installation via bootloader went fine using a Raspberry Pi running Debian Buster (no bootloader access on Windows PCs).
Averaging works a lot better now :)

Minor observation:
- My statistics data and setup parameters from the previous firmware (RadPro 2.0.1) got overwritten and replaced by default values.

**Reply by Gissio** [reply]
Did you use the -update script?

**Reply by dc1rdb** [reply]
No, I just did a drag and drop of the .bin file onto the GC-01 bootloader drive via Debian desktop. Had to do it twice, but that is a known quirk.

**Reply by Gissio** [reply]
I forgot to upload the bootloader update versions. Thanks for your report!

**Comment by Gissio** [comment]
For those curious about how averaging works now in Rad Pro 2.0.2, here's a breakdown:

## Adaptive fast

* This option averages the last 20 pulses, providing a confidence interval of ±50%. This helps avoid the loss of precision sometimes observed with the previous algorithm.
* With M4011/J321/J302/SBM-20 tubes, the response time for 150 cpm (~1 µSv/h) is 8 seconds.
* With J613/J614 tubes, the response time for 60 cpm (~1 µSv/h) is 20 seconds.
* You can view the response curves here (radiation level vs. response time, for M4011 tubes): [instantaneousaveraging-adaptivefast.ipynb](https://github.com/Gissio/radpro/blob/main/tests/instantaneousaveraging-adaptivefast.ipynb)

## Adaptive precision

* Similar to "Adaptive fast", this option also sets a minimum averaging time of 5 seconds.
* This ensures higher precision when more than 20 pulses occur within the last 5 seconds. Otherwise it functions identically to "Adaptive fast".
* You can view the response curves here (radiation level vs. response time, for M4011 tubes): [instantaneousaveraging-adaptiveprecision.ipynb](https://github.com/Gissio/radpro/blob/main/tests/instantaneousaveraging-adaptiveprecision.ipynb)
